api/src/neo4j/queries/map.js

```diff
@@ -74,7 +106,10 @@ RETURN svg { id: svg.id, name: svg.customName, svgs: [svg {.*}]}
     customs,
   };
 
-  return mapListing;
+  const compartments = await mapComponents(mapListing.compartments);
+  const subsystems = await mapComponents(mapListing.subsystems);
+
+  return {...mapListing, compartments, subsystems };
```

Comment:
So just to clarify, you would like to replace this with doing a request to get the amount of reactions the  compartment/subsystem has which the map belongs to? I think that's not too complicated but since this does not seem to be too big of an issue I'm fine with keeping it as it is

Reply:
How about we keep it as is and open a (refactoring) issue?